SofaKernel/modules/SofaBaseMechanics/SofaBaseMechanics_test/DiagonalMass_test.cpp

```diff
@@ -646,6 +651,351 @@ class DiagonalMass_test : public BaseTest
+        EXPECT_NEAR(mass->getTotalMass(), 0, 1e-4);
+        
+        return;
+    }
+
+
+    void checkTopologicalChanges_Tetra()
+    {
+        string scene =
+            "<?xml version='1.0'?>                                                                              "
+            "<Node  name='Root' gravity='0 0 0' time='0' animate='0'   >                                        "
+            "    <RequiredPlugin name='SofaTopologyMapping'/>                                                   "
+            "    <RegularGridTopology name='grid' n='2 2 2' min='0 0 0' max='2 2 2' p0='0 0 0' />               "
+            "    <Node name='Tetra' >                                                                           "
+            "            <MechanicalObject position='@../grid.position' />                                      "
+            "            <TetrahedronSetTopologyContainer name='Container' />                                   "
+            "            <TetrahedronSetTopologyModifier name='Modifier' />                                     "
+            "            <TetrahedronSetGeometryAlgorithms template='Vec3d' name='GeomAlgo' />                  "
+            "            <Hexa2TetraTopologicalMapping input='@../grid' output='@Container' />                  "
+            "            <DiagonalMass name='m_mass' massDensity='1.0'/>                                        "
+            "    </Node>                                                                                        "
+            "</Node>                                                                                            ";
+
+        Node::SPtr root = SceneLoaderXML::loadFromMemory("loadWithNoParam",
+            scene.c_str(),
+            sofa::Size(scene.size()));
+        ASSERT_NE(root.get(), nullptr);
+        
+        /// Init simulation
+        sofa::simulation::getSimulation()->init(root.get());
+
+        TheDiagonalMass* mass = root->getTreeObject<TheDiagonalMass>();
+        ASSERT_NE(mass, nullptr);
+
+        TetrahedronSetTopologyModifier* modifier = root->getTreeObject<TetrahedronSetTopologyModifier>();        
+        ASSERT_NE(modifier, nullptr);
+
+        const VecMass& vMasses = mass->d_vertexMass.getValue();
+        Real refValue = Real(1.0/3.0);  //0.3333
+        Real refValue2 = 2 - refValue; // 1.6667
+       
+        // check value at init
+        EXPECT_EQ(vMasses.size(), 8);
+        EXPECT_NEAR(vMasses[0], refValue2, 1e-4);
+        EXPECT_NEAR(mass->getTotalMass(), 8.0, 1e-4);
+
+        sofa::helper::vector<sofa::Index> tetraIds = { 0 };
+        // remove tetrahedron id: 0
+        modifier->removeTetrahedra(tetraIds); 
+        EXPECT_EQ(vMasses.size(), 8);
+        EXPECT_NEAR(vMasses[0], refValue2 - refValue, 1e-4); // check update of Mass when removing tetra
+        EXPECT_NEAR(mass->getTotalMass(), 8.0 - (4 * refValue), 1e-4);
+        Real lastV = vMasses[7];
```

Comment:
```suggestion
        const Real lastV = vMasses[7];
```
Note that it could also be `vMasses.back()`